File: typescript/resistor-color-duo/resistor-color-duo.ts
const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];

export class ResistorColor {
  private colors: string[];

  constructor(colors: string[]) {
    if (!colors || colors.length < 2) {
      throw "At least two colors need to be present";
    }
    this.colors = colors;
  }
  value = (): number => {
    return parseInt(
      (COLORS.findIndex((x) => x === this.colors[0])?.toString() ?? "0") +
        (COLORS.findIndex((x) => x === this.colors[1])?.toString() ?? "0")
    );
  };
}

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

class ResistorColorDuo {
    private static final List<String> COLORS = List.of(
            "black",
            "brown",
            "red",
            "orange",
            "yellow",
            "green",
            "blue",
            "violet",
            "grey",
            "white"
    );


    int value(String[] colors) {
        final var colorIndexes = Arrays.stream(colors)
                .limit(2)
                .map(COLORS::indexOf)
                .toList();

        int result = 0;

        for (int i = 0; i < colorIndexes.size(); i++) {
            result += (int) (colorIndexes.get(i) * Math.pow(10.0, colorIndexes.size() - i - 1));
        }

        return result;
    }
}
